monai/transforms/post/array.py

```diff
@@ -122,94 +122,68 @@ class AsDiscrete(Transform):
     Args:
         argmax: whether to execute argmax function on input data before transform.
             Defaults to ``False``.
-        to_onehot: whether to convert input data into the one-hot format.
-            Defaults to ``False``.
-        num_classes: the number of classes to convert to One-Hot format.
+        to_onehot: if not None, convert input data into the one-hot format with specified number of classes.
+            Defaults to ``None``.
+        threshold: if not None, threshold the float values to int number 0 or 1 with specified theashold.
```

Comment:
forgot the `threshold=False` case, should it be allowed (convert to None) or raise value error?

Reply:
Hi @wyli ,

In that case, should we also add error message for `to_onehot=True/False`? That's why I said there are too many deprecated args here and making the code confusing.

Thanks.

Reply:
Anyway, let me add the logic to improve the experience for users.

Thanks.